core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1214,16 +1214,20 @@ private void launchTcpSlaveAgentListener() throws IOException {
                 tcpSlaveAgentListener = null;
             }
             if (slaveAgentPort != -1 && tcpSlaveAgentListener == null) {
-                String administrativeMonitorId = getClass().getName() + ".tcpBind";
+                final String administrativeMonitorId = getClass().getName() + ".tcpBind";
                 try {
                     tcpSlaveAgentListener = new TcpSlaveAgentListener(slaveAgentPort);
                     // remove previous monitor in case of previous error
-                    for (Iterator<AdministrativeMonitor> it = AdministrativeMonitor.all().iterator(); it.hasNext(); ) {
+                    AdministrativeMonitor toBeRemoved = null;
+                    ExtensionList<AdministrativeMonitor> all = AdministrativeMonitor.all();
+                    for (Iterator<AdministrativeMonitor> it = all.iterator(); it.hasNext(); ) {
```

Comment:
Switch to Java 5 foreach, then, as you don't need anymore the old-school iterator style?